emacs: fails to launch on macOS Sequoia 15.4

### Nixpkgs version

- Unstable (25.05)

### Describe the bug

On current `nipxkgs-unstable` (88e992074d86ad50249de12b7fb8dbaadf8dc0c5), after updating my Mac to Sequoia 15.4, any variant of Emacs fails to launch.

Here's `emacs-pgtk`, which was in my Nix store:

```
$ nix build -L .#legacyPackages.aarch64-darwin.emacs-pgtk
$ ./result/Applications/Emacs.app/Contents/MacOS/Emacs
dyld[8612]: Library not loaded: /nix/store/lzxyfrbxm4vjxya4fxvibnw3pj4fqsfn-libgccjit-14-20241116/lib/libgccjit.0.dylib
  Referenced from: <E60A2E51-AB70-3736-B2D4-E95A42A38F84> /nix/store/cf2q39fihv3ds6n77p7lbjgg0h0sry4j-emacs-pgtk-30.1/Applications/Emacs.app/Contents/MacOS/Emacs
  Reason: tried: '/nix/store/lzxyfrbxm4vjxya4fxvibnw3pj4fqsfn-libgccjit-14-20241116/lib/libgccjit.0.dylib' (duplicate LC_RPATH '@loader_path'), '/System/Volumes/Preboot/Cryptexes/OS/nix/store/lzxyfrbxm4vjxya4fxvibnw3pj4fqsfn-libgccjit-14-20241116/lib/libgccjit.0.dylib' (no such file), '/nix/store/lzxyfrbxm4vjxya4fxvibnw3pj4fqsfn-libgccjit-14-20241116/lib/libgccjit.0.dylib' (duplicate LC_RPATH '@loader_path'), '/usr/local/lib/libgccjit.0.dylib' (no such file), '/usr/lib/libgccjit.0.dylib' (no such file, not in dyld cache)
[1]    8612 abort      ./result/Applications/Emacs.app/Contents/MacOS/Emacs
```

`open /Applications/Nix\ Apps/Emacs.app` results in a similar error as seen in Console.app:

```
ASI found [dyld] (sensitive) 'Library not loaded: /nix/store/lzxyfrbxm4vjxya4fxvibnw3pj4fqsfn-libgccjit-14-20241116/lib/libgccjit.0.dylib
  Referenced from: <E60A2E51-AB70-3736-B2D4-E95A42A38F84> /nix/store/cf2q39fihv3ds6n77p7lbjgg0h0sry4j-emacs-pgtk-30.1/Applications/Emacs.app/Contents/MacOS/Emacs
  Reason: tried: '/nix/store/lzxyfrbxm4vjxya4fxvibnw3pj4fqsfn-libgccjit-14-20241116/lib/libgccjit.0.dylib' (duplicate LC_RPATH '@loader_path'), '/System/Volumes/Preboot/Cryptexes/OS/nix/store/lzxyfrbxm4vjxya4fxvibnw3pj4fqsfn-libgccjit-14-20241116/lib/libgccjit.0.dylib' (no such file), '/nix/store/lzxyfrbxm4vjxya4fxvibnw3pj4fqsfn-libgccjit-14-20241116/lib/libgccjit.0.dylib' (duplicate LC_RPATH '@loader_path'), '/usr/local/lib/libgccjit.0.dylib' (no such file), '/usr/lib/libgccjit.0.dylib' (no such file, not in dyld cache)'
```

Attempts to rebuild Emacs result in a related failure (slightly older version of `nixpkgs-unstable` used here, for reasons):

```
emacs-default> Running phase: unpackPhase
emacs-default> unpacking source archive /nix/store/r98nampli716xi1z89c59a2z9rym3lx5-default.el
emacs-default> source root is .
emacs-default> Running phase: patchPhase
emacs-default> Running phase: updateAutotoolsGnuConfigScriptsPhase
emacs-default> Running phase: configurePhase
emacs-default> no configure script, doing nothing
emacs-default> Running phase: buildPhase
emacs-default> dyld[8955]: Library not loaded: /nix/store/lzxyfrbxm4vjxya4fxvibnw3pj4fqsfn-libgccjit-14-20241116/lib/libgccjit.0.dylib
emacs-default>   Referenced from: <E60A2E51-AB70-3736-B2D4-E95A42A38F84> /nix/store/cf2q39fihv3ds6n77p7lbjgg0h0sry4j-emacs-pgtk-30.1/bin/.emacs-30.1-wrapped
emacs-default>   Reason: tried: '/nix/store/lzxyfrbxm4vjxya4fxvibnw3pj4fqsfn-libgccjit-14-20241116/lib/libgccjit.0.dylib' (duplicate LC_RPATH '@loader_path'), '/usr/local/lib/libgccjit.0.dylib' (no such file), '/usr/lib/libgccjit.0.dylib' (no such file, not in dyld cache)
emacs-default> /nix/store/hjih3ih7wdq3cgqrlz3ry6dnbdjn04qv-stdenv-darwin/setup: line 1777:  8955 Abort trap: 6           emacs -l package -f package-initialize -L . --batch -f batch-byte-compile *.el
error: builder for '/nix/store/nbwnr3wh5hgfa0vs8j07c6fm2w03vsfa-emacs-default-0.1.0.drv' failed with exit code 134;
       last 12 log lines:
       > Running phase: unpackPhase
       > unpacking source archive /nix/store/r98nampli716xi1z89c59a2z9rym3lx5-default.el
       > source root is .
       > Running phase: patchPhase
       > Running phase: updateAutotoolsGnuConfigScriptsPhase
       > Running phase: configurePhase
       > no configure script, doing nothing
       > Running phase: buildPhase
       > dyld[8955]: Library not loaded: /nix/store/lzxyfrbxm4vjxya4fxvibnw3pj4fqsfn-libgccjit-14-20241116/lib/libgccjit.0.dylib
       >   Referenced from: <E60A2E51-AB70-3736-B2D4-E95A42A38F84> /nix/store/cf2q39fihv3ds6n77p7lbjgg0h0sry4j-emacs-pgtk-30.1/bin/.emacs-30.1-wrapped
       >   Reason: tried: '/nix/store/lzxyfrbxm4vjxya4fxvibnw3pj4fqsfn-libgccjit-14-20241116/lib/libgccjit.0.dylib' (duplicate LC_RPATH '@loader_path'), '/usr/local/lib/libgccjit.0.dylib' (no such file), '/usr/lib/libgccjit.0.dylib' (no such file, not in dyld cache)
       > /nix/store/hjih3ih7wdq3cgqrlz3ry6dnbdjn04qv-stdenv-darwin/setup: line 1777:  8955 Abort trap: 6           emacs -l package -f package-initialize -L . --batch -f batch-byte-compile *.el
       For full logs, run 'nix log /nix/store/nbwnr3wh5hgfa0vs8j07c6fm2w03vsfa-emacs-default-0.1.0.drv'.
error: 1 dependencies of derivation '/nix/store/iq009riiqlag59ih46jrfrn6f8hvahsl-emacs-pgtk-with-packages-30.1.drv' failed to build
error: 1 dependencies of derivation '/nix/store/0zdnjin2fl6rbjrpjrjjyzkvdhkhw10j-system-applications.drv' failed to build
error: 1 dependencies of derivation '/nix/store/2k3qw8xc3ajky3bwhi6rqbgwg7gg79cd-darwin-system-25.05.75f8e4d.drv' failed to build
```

### Steps to reproduce

1. Upgrade to macOS Sequoia 15.4.
2. Build `emacs-pgtk`, or run an existing binary.

### Expected behaviour

Emacs launches/builds successfully.

### Screenshots

_No response_

### Relevant log output

```console

```

### Additional context

Note that I'm using the sandbox. Relevant `nix.conf`:

```
sandbox = true
sandbox-fallback = false
```

### System metadata

 - system: `"aarch64-darwin"`
 - host os: `Darwin 24.4.0, macOS 15.4`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.12`
 - nixpkgs: `/nix/store/n5zhvsrwjrlyaimd3cp7s70j401iycx5-source`


### Are you using nix-darwin?

Yes, I am using nix-darwin.

### Notify maintainers

@AndersonTorres @adisbladis @jwiegley @lovek323 @matthewbauer @panchoh @NixOS/darwin-core

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+bug%22+label%3A%226.topic%3A+darwin%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

Hi everybody,
could someone help me? I had sucesfully used the Workaround to build emacs on MacOS 15.4
```
 programs.emacs = {
    enable = true;
    package = emacsPackage.override {
      withNativeCompilation = false;
    };
```

But somehow my build is failing again:

```
nix build .#darwinConfigurations.MB-FVFH2468Q0.system --extra-experimental-features 'nix-command flakes'
error: builder for '/nix/store/yi61xg3mlxhdgmq1zna15apkshwn0cip-emacs-30.1.drv' failed with exit code 134;
       last 25 log lines:
       > checking for getaddrinfo_a in -lanl... no
       > checking for malloc_trim... no
       > checking for gnutls >= 2.12.2... yes
       > checking for libsystemd >= 222... no
       > checking for tree-sitter >= 0.20.2... yes
       > checking for sys/inotify.h... no
       > checking for libkqueue... no
       > checking for library containing kqueue... none required
       > checking for jpeglib 6b or later... -ljpeg
       > checking for lcms2... no
       > checking for library containing inflateEnd... -lz
       > checking for dladdr... yes
       > checking for dlfunc... no
       > checking for gcc_jit_context_acquire in -lgccjit... yes
       > checking for libgccjit.h... yes
       > configure: error: The installed libgccjit failed to compile and run a test program using
       > the libgccjit library; see config.log for the details of the failure.
       > The test program can be found here:
       > <https://gcc.gnu.org/onlinedocs/jit/intro/tutorial01.html>.
       > You can try compiling it yourself to investigate the issues.
       > Please report the issue to your distribution if libgccjit was installed
       > through that.
       > You can find the instructions on how to compile and install libgccjit from
       > source on this site:
       > <https://gcc.gnu.org/wiki/JIT>.
       For full logs, run 'nix log /nix/store/yi61xg3mlxhdgmq1zna15apkshwn0cip-emacs-30.1.drv'.
error: 1 dependencies of derivation '/nix/store/y39n5zpficsg7zy3yndhn3048c26xjga-activation-script.drv' failed to build
error: 1 dependencies of derivation '/nix/store/3qyhpiyrg2q7gkprraxpadwkhwxg64sd-home-manager-generation.drv' failed to build
error: 1 dependencies of derivation '/nix/store/r4jmgkc68b1c9q67548vz2aji4i72v9a-activation-TMUNDIN.drv' failed to build
error (ignored): error: cannot unlink '"/private/tmp/nix-build-emacs-30.1.drv-5/source"': Directory not empty
error: 1 dependencies of derivation '/nix/store/m89hj4s7srjdyjrn80mhkwh8j0my3vwg-darwin-system-25.05.73d5958.drv' failed to build
error: Recipe `darwin` failed on line 14 with exit code 1
```

> Hi everybody, could someone help me? I had sucesfully used the Workaround to build Emacs on macOS 15.4
> 
> ```
>  programs.emacs = {
>     enable = true;
>     package = emacsPackage.override {
>       withNativeCompilation = false;
>     };
> ```
> 
> But somehow my build is failing again:
> 
> ```
> nix build .#darwinConfigurations.MB-FVFH2468Q0.system --extra-experimental-features 'nix-command flakes'
> error: builder for '/nix/store/yi61xg3mlxhdgmq1zna15apkshwn0cip-emacs-30.1.drv' failed with exit code 134;
>        last 25 log lines:
>        > checking for getaddrinfo_a in -lanl... no
>        > checking for malloc_trim... no
>        > checking for gnutls >= 2.12.2... yes
>        > checking for libsystemd >= 222... no
>        > checking for tree-sitter >= 0.20.2... yes
>        > checking for sys/inotify.h... no
>        > checking for libkqueue... no
>        > checking for library containing kqueue... none required
>        > checking for jpeglib 6b or later... -ljpeg
>        > checking for lcms2... no
>        > checking for library containing inflateEnd... -lz
>        > checking for dladdr... yes
>        > checking for dlfunc... no
>        > checking for gcc_jit_context_acquire in -lgccjit... yes
>        > checking for libgccjit.h... yes
>        > configure: error: The installed libgccjit failed to compile and run a test program using
>        > the libgccjit library; see config.log for the details of the failure.
>        > The test program can be found here:
>        > <https://gcc.gnu.org/onlinedocs/jit/intro/tutorial01.html>.
>        > You can try compiling it yourself to investigate the issues.
>        > Please report the issue to your distribution if libgccjit was installed
>        > through that.
>        > You can find the instructions on how to compile and install libgccjit from
>        > source on this site:
>        > <https://gcc.gnu.org/wiki/JIT>.
>        For full logs, run 'nix log /nix/store/yi61xg3mlxhdgmq1zna15apkshwn0cip-emacs-30.1.drv'.
> error: 1 dependencies of derivation '/nix/store/y39n5zpficsg7zy3yndhn3048c26xjga-activation-script.drv' failed to build
> error: 1 dependencies of derivation '/nix/store/3qyhpiyrg2q7gkprraxpadwkhwxg64sd-home-manager-generation.drv' failed to build
> error: 1 dependencies of derivation '/nix/store/r4jmgkc68b1c9q67548vz2aji4i72v9a-activation-TMUNDIN.drv' failed to build
> error (ignored): error: cannot unlink '"/private/tmp/nix-build-emacs-30.1.drv-5/source"': Directory not empty
> error: 1 dependencies of derivation '/nix/store/m89hj4s7srjdyjrn80mhkwh8j0my3vwg-darwin-system-25.05.73d5958.drv' failed to build
> error: Recipe `darwin` failed on line 14 with exit code 1
> ```

Maybe interesting for all who face the same Error. I found the Problem!
If you have a `home.activation.` defined the build is failing. I don't get why this happen. 
This is my Code. Was tryng to automatically sync doom emacs:
```
  home.activation.doom-sync = lib.hm.dag.entryAfter ["writeBoundary"] ''
    echo "Doom sync..."
    if [[ -f ${config.xdg.configHome}/emacs/bin/doom ]]; then
      export PATH="${emacsPackage}/bin:${pkgs.git}/bin:$PATH"
      ${config.xdg.configHome}/emacs/bin/doom sync
      ${config.xdg.configHome}/emacs/bin/doom env
    fi
  '';
```

--- a/pkgs/applications/editors/emacs/make-emacs.nix
+++ b/pkgs/applications/editors/emacs/make-emacs.nix
@@ -67,11 +67,7 @@
   zlib,
 
   # Boolean flags
-
-  # FIXME: Native compilation breaks build and runtime on macOS 15.4;
-  # see <https://github.com/NixOS/nixpkgs/issues/395169>.
-  withNativeCompilation ?
-    stdenv.buildPlatform.canExecute stdenv.hostPlatform && !stdenv.hostPlatform.isDarwin,
+  withNativeCompilation ? stdenv.buildPlatform.canExecute stdenv.hostPlatform,
   noGui ? false,
   srcRepo ? true,
   withAcl ? false,

